src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -569,8 +569,15 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
       throw getMatAutocompleteMissingPanelError();
     }
 
-    if (!this._overlayRef) {
+    if (!this._portal || this._portal.templateRef !== this.autocomplete.template) {
```

Comment:
A cleaner approach to this would be to move the `_portal` to `MatAutocomplete` and have the trigger just call `OverlayRef.attach`.

Reply:
I would've preferred to move it for the menu as well, but it has a different set of requirements, because the consumer can pass in any object that implements the `MatMenuPanel` interface.

Reply:
I moved the portal to the `mat-autocomplete`. But we still have to detach first in the trigger before attaching a new portal, otherwise it will throw the "already attached" error.
I am also un- and resubscribing to closing actions now when attaching a new portal, so we get the correct option changes events.